### PR TITLE
[WIP] Thor binary

### PR DESCRIPTION
**Pull request by wvanbergen**
cc @dalehamel @gedge

### CODE REVIEW

File: bin/terraform_dsl [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+begin
+  ENV["THOR_DEBUG"] = "1"
+  Terraform::CLI.start(ARGV)
+rescue Thor::UndefinedCommandError, Thor::UnknownArgumentError, Thor::AmbiguousCommandError, Thor::InvocationError => e
+  $stderr.puts(e.message)
+  exit(64)
+rescue Thor::Error => e
+  $stderr.puts(e.message)
+  exit(1)
+end
```
Comment (wvanbergen) [inline bin/terraform_dsl:14]: Normally thor always exits with exit status 0, which is shitty when building scripts. The debug options allows you to handle exceptions yourself.

**Comment by dalehamel** [comment]
It's a good start, we'll want pull out and DRY up the code in terraform.rake, and then delete lib/tasks.

**Comment by wvanbergen** [comment]
Closing this; will reopen a new PR soon

**Comment by gedge** [comment]
@wvanbergen - think you meant @thegedge (not me)

**Comment by wvanbergen** [comment]
Whoops :)

**Pull request closed without merging**